File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.py
def solution(babbling):
    
    answer = 0
    memories = ["aya", "ye", "woo", "ma"]
      
    for word in babbling:
        for memory in memories:
            if memory * 2 not in word:
                word = word.replace(memory, ' ')
        if word.isspace():
            
            answer += 1 
        
    return answer

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.js
function solution(babbling) {
    var words = ["aya", "ye", "woo", "ma"];
    var canSpeaking = [];
    for(var i=0; i<words.length; i++) {
        babbling = babbling.map((a) => a.replaceAll(words[i],i));
    }
    var current = "";
    var count = 0;
    for(var bab of babbling) {
        var check = true;
        var cur = "";
        for(var c of [...bab]) {
            if (isNaN(c)) {
                check = false;
                break;
            } else if (cur === "") {
                cur = c;
            } else if (cur == c) {
                check = false;
                break;
            } else {
                cur = c;
            }
        }
        
        if (check) {
            count += 1;
        }
    }
    return count;
}